subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -482,6 +492,19 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t remainder, uint16_t laz
 	struct ll_scan_set *scan = param;
 	uint32_t ret;
 	uint8_t ref;
+	struct lll_scan *lll;
```

Comment:
This is incorrect, you have to design a "periodic scanning" for a "duration". You are shunting the "scan interval" after "duration" in your implementation here.

Reply:
Yes, I'm shunting it until start of the next period, so it will scan for "duration", restarting every "period".

Reply:
By shunting, other BLE states/roles are blocked from using the radio air time as scanner is reserving the window every interval, beats the purpose of period and duration feature.

Reply:
Modified code to avoid this